### PR TITLE
feat(windows): convert locale.xml to strings.xml format

### PR DESCRIPTION
**Pull request by mcdurdin**
Updates all references to locale.xml and strings within to use the Android strings.xml format.

Half of the changes are the transfer from locale.xml to strings.xml (diverges enough that it's not detected as a rename by git). I intentionally kept the files as similar as possible so a line-by-line comparison is possible if you are very keen).

* I have built and tested Keyman on my machine and appears to be working well.
* I have not yet build FirstVoices Keyboards and there will be additional cleanup to do there (old files to remove, strings.xml conversion to do). I think I will do the rest of that in a separate PR.

### CODE REVIEW

File: windows/src/global/delphi/ui/XMLRenderer.pas [hunk 1]
```diff
@@ -215,7 +215,6 @@ function TXMLRenderers.RenderToString(FRefreshKeyman: Boolean; const AdditionalD
         IfThen(CanElevate, '<canelevate />')+
         IfThen(Assigned(kmcom) and kmcom.SystemInfo.IsAdministrator, '<isadmin />')+   // I3612   // I3626
         '<uilanguages>'+GetUILanguages+'</uilanguages>'+
-        '<defaultlocalepath>'+XMLEncode(FTemplatePath)+'locale.xml</defaultlocalepath>'+
         '<localepath>'+XMLEncode(s)+'</localepath>');
```
Comment (darcywong00) [inline windows/src/global/delphi/ui/XMLRenderer.pas:218]: Checking `<defaultlocalepath>` was no longer needed

Comment (mcdurdin) [reply windows/src/global/delphi/ui/XMLRenderer.pas:218]: Correct -- it's now always `strings.xml` in the same path as the template, so no path information needed (in the past this was really messy, but improved with internal web server).

File: windows/src/desktop/kmshell/xml/splash.xsl [hunk 2]
```diff
@@ -49,7 +49,7 @@
   </div>
 
   <div id="version"><xsl:value-of select="/Keyman/Version"/>
-    <span id="beta"><xsl:value-of select="$locale/String[@Id='S_BETA']"/></span>
+    <span id="beta"><xsl:value-of select="$locale/string[@name='S_BETA']"/></span>
```
Comment (darcywong00) [inline windows/src/desktop/kmshell/xml/splash.xsl:52]: 1. I was expecting to find `S_BETA` in strings.xml but I didn't find it.
2. Will there also be an entry for `S_ALPHA`?

File: windows/src/desktop/locale/locale-to-android-strings.xsl [hunk 3]
```diff
@@ -5,7 +5,9 @@
 	<xsl:template match="String">
     <string>
       <xsl:attribute name="name"><xsl:value-of select="@Id" /></xsl:attribute>
-      <xsl:attribute name="comment"><xsl:value-of select="@Group" />/<xsl:value-of select="@Type" />: <xsl:value-of select="@Description" /></xsl:attribute>
+      <xsl:attribute name="comment"><xsl:value-of select="@Group" />/<xsl:value-of select="@Type" />: <xsl:value-of select="@Description" 
+        /><xsl:if test="@Parameters"> [<xsl:value-of select="@Parameters" />]</xsl:if
+        ><xsl:if test="@Version"> (introduced <xsl:value-of select="@Version" />)</xsl:if></xsl:attribute>
```
Comment (darcywong00) [inline windows/src/desktop/locale/locale-to-android-strings.xsl:10]: Do you want to handle the attribute `translatable`?
An example from KMEA strings.xml
```
    <!-- Shared preferences name -->
    <string name="kma_prefs_name" translatable="false">KMAPreferences</string>
```

File: windows/src/desktop/kmshell/xml/splash.xsl (hunk 2 above)
Comment (mcdurdin) [reply windows/src/desktop/kmshell/xml/splash.xsl:52]: This `S_BETA` reference is deleted in the next PR (#3428)

File: windows/src/desktop/locale/locale-to-android-strings.xsl (hunk 3 above)
Comment (mcdurdin) [reply windows/src/desktop/locale/locale-to-android-strings.xsl:10]: I think all strings listed are translatable 😁  (also, these xsl files will be unnecessary once we complete the transition).

**Pull request merged**